sdk/src/main/java/com/ifountain/opsgenie/client/AbstractOpsGenieHttpClient.java

```diff
@@ -126,6 +126,57 @@ protected BaseResponse doGetRequest(BaseRequest request) throws OpsGenieClientEx
         return populateResponse(request, httpResponse);
     }
 
+    protected BaseResponse doPostRequestV2(BaseRequest request) throws IOException, OpsGenieClientException, ParseException {
+        setAndValidateApiKey(request);
+        Map<String, String> headers = request.getReqHeadersForPostOrPatch();
+        String uri = rootUri + request.getEndPoint();
+        String json = JsonUtils.toJson(request);
+        log.info("Executing OpsGenie request to [" + uri + "] with content Parameters:" + LogUtils.getInsensitiveLogMessage(JsonUtils.toMap(request)));
+        OpsGenieHttpResponse httpResponse = httpClient.post(uri, json, headers);
+        handleResponse(httpResponse);
+        return populateResponse(request, httpResponse);
+    }
+    protected BaseResponse doGetRequestV2(BaseRequest request) throws OpsGenieClientException, IOException, ParseException {
+        setAndValidateApiKey(request);
+        String uri = rootUri + request.getEndPoint();
+
+        Map<String,Object> parameters = request.getRequestParams();
```

Comment:
Almost complete function for get and delete is the same. Can we abstract it to another function? Same is the case between post and patch.

Reply:
Changes done